tools/ncf.py

```diff
@@ -154,6 +156,9 @@ def parse_bundlefile_metadata(content, bundle_type):
         # tag "parameter" may be multi-valued
         if tag == "parameter":
           parameters.append({'name': match.group(3), 'description': match.group(4)})
+        if tag == "parameter_constraint":
+          constraint = json.loads("{" + match.group(4)+ "}")
+          param_constraints.append({'name': match.group(3), 'constraint': constraint})
```

Comment:
wouldn't it be easier if you used a dict name=>constraints for param_constraints ? 
Or do you allow multiple constraint line for a single parameter ?